### ISSUE TITLE
Example usage in the README ?

### ISSUE THREAD

**Issue by stuaxo** [jschwartzenberg/dos-download-scripts#1]
Hi,
   Do these work yet ?   Any chance of documenting usage in the README ?

S

**Comment by jschwartzenberg** [comment]
They all work, but there isn't much error handling. It's all made for the happy flow scenario. I use these mostly for working and testing things with dosemu.

The `dostricks` script accepts two arguments, first a destination and the second is a filename from the `tricks` directory. It will download the ZIPs for all the disk images and extract them and then for each disk image from the ZIPs extract that again into the directory supplied as the destination argument. For all of these, the next step then is just running the installer from DOSEMU as all the files it expects will be there.

The `install-*` scripts use OCR (gocr) to read the text inside the DOSEMU window and perform key presses to do installations automatically. Unfortunately gocr is not that good, so the strings it checks are really tweaked for that. Here is a video of it running properly:
https://schwart6.home.xs4all.nl/dosemu-msoffice.webm

I hope that it would be possible to find a good and fast way to do the OCR, it would be interesting to build a testing framework around that.

I should probably document these indeed, this is the first time I notice there are others interested :)

**Comment by stuaxo** [comment]
Interesting, not much comment until I give these a go.

The OCR might improve if you can add examples of the VGA font, and the fonts windows 3.1 uses, if you haven't already.

**Comment by jschwartzenberg** [comment]
No, I did not do that. That seems interesting. Are you familiar with gocr?

I found that the most recent version regressed a lot compared to the previous one when I tried it. When I mailed the author, he told me he would be doing a new release that would fix a lot of issues, but I never saw anything anymore.

**Comment by stuaxo** [comment]
I'm not familiar with GOCR, but I did work on a project for a while that used OCR for automation (the OCR wasn't the code I worked with though). 

I'll have a look at GOCR. 


OT:
While poking around, I found this tool to automate installation of windows 31 for dosbox

http://www.columbia.edu/~em36/win31dosbox.html

**Comment by stuaxo** [comment]
As an alternative to OCR, SikuliX looks interesting

sikulix.com

**Comment by jschwartzenberg** [comment]
That automated installation is interesting. That's indeed what I was thinking of moving to, but it's great to have this example for DOSBox, thanks!

I looked at SikuliX, it seems to be a Java wrapper around Tesseract for OCR. With Java it would seem nice to be able to use in combination with Cucumber/Serenity to write test scenarios. If I'd have time for that I'd look into setting something up to automate DOSEMU testing :)
